src/prefect/engine/results/gcs_result.py

```diff
@@ -15,21 +15,22 @@ class GCSResult(Result):
     runtime environment has the proper credentials available
     (see https://cloud.google.com/docs/authentication/production for all the authentication options).
 
-    You can also optionally provide the name of a Prefect Secret containing your
-    service account key. To read more about service account keys see https://cloud.google.com/iam/docs/creating-managing-service-account-keys.
+    You can also optionally provide your service account key to `prefect.context.secrets.GCP_CREDENTIALS` for
+    automatic authentication - see [Third Party Authentication](../../../orchestration/recipes/third_party_auth.html) for more information.
+
+    To read more about service account keys see https://cloud.google.com/iam/docs/creating-managing-service-account-keys.
     To read more about the JSON representation of service account keys see https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts.keys.
 
     Args:
         - bucket (str): the name of the bucket to write to / read from
-        - credentials_secret (str, optional): the name of the Prefect Secret
-            which stores a JSON representation of your Google Cloud credentials.
         - **kwargs (Any, optional): any additional `Result` initialization options
     """
 
     def __init__(
         self, bucket: str = None, credentials_secret: str = None, **kwargs: Any
     ) -> None:
         self.bucket = bucket
+        self.logger = logging.get_logger(type(self).__name__)
```

Comment:
The base result class [already has a logger](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/result/base.py#L113)